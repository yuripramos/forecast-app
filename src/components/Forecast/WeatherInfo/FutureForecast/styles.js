import styled, { css } from "styled-components";
import { rem, media } from "../../../../styles/tools";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 100%;
  display: inline-block;
`;
export const InnerContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(250)};
  display: inline-flex;
`;


export const Icon = styled.div`
  width: 100%;
  height: ${rem(60)};
`;

export const DaySpecs = styled.div`
  margin: ${rem(5)} ${rem(13)} ${rem(10)} ${rem(13)};
`;

export const Temperature = styled.h1`
  font-size: 1.5em;
  font-weight: 200;
  line-height: 0.75;
  font-family: "RobotoSlab Thin";
  margin-top: ${rem(20)};
  width: 100%;
   ${({ tiny }) =>
    tiny &&
    css`
      font-size: 1em;
  `};
`;

export const TinySpecs = styled.h3`
  color: white;
  font-weight: 100;
  line-height: 1.1;
  letter-spacing: 0.025em;
  margin: 0;
  padding: 0;
  font-family: "OpenSans Thin";
  width: 100%;
  margin: ${rem(10)} 0;
  :nth-child(2n) {
    margin: ${rem(2)} 0;
  }
`;
