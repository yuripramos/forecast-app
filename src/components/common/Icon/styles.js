import styled, { css } from "styled-components";
import { rem } from "../../../styles/tools/index";

export const Span = styled.span`
  display: inline-flex;

  ${({ color }) =>
    color &&
    css`
      color: ${color};
    `};

  ${({ width, height }) =>
    width &&
    height &&
    css`
      svg {
        width: ${rem(width)};
        height: ${rem(height)};
      }
    `};
`;
