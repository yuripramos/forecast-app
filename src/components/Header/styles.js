import styled, { css } from "styled-components";
import { rem, media } from "../../styles/tools";
import { white, trueBlack, lightBlue } from "../../styles/settings";

export const HeaderWrapper = styled.header`
  height: ${rem(158)};
  background: ${lightBlue};
  padding-top: ${rem(10)};
  text-align: center;
  ${media.md(css`
    text-align: center;
  `)};
  ${media.lg(css`
    height: ${rem(260)};
    padding-top: ${rem(6)};
  `)};
`;

export const Title = styled.h1`
  font-family: "Pacifico";
  font-size: ${rem(45)};
  text-align: center;
  color: ${white};
  text-shadow: ${rem(3)} ${rem(3)} rgba(0, 0, 0, 0.25);
  margin-top: ${rem(-5)};
  ${media.md(css`
    text-align: left;
  `)};
`;

export const SubTitle = styled.h2`
  font-family: "Helveticaneue Medium";
  font-size: ${rem(20)};
  text-align: center;
  color: ${white};
  text-shadow: ${rem(3)} ${rem(3)} rgba(0, 0, 0, 0.25);
  ${media.md(css`
    font-size: ${rem(24)};
    text-align: left;
  `)};
`;

export const IconWrapper = styled.div`
  width: ${rem(70)};
  height: ${rem(55)};
  background: ${trueBlack};
  border-radius: ${rem(40)};
  opacity: 0.5;
  margin: ${rem(5)};
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  :hover {
    opacity: 1;
  }
`;

export const ListIcons = styled.div`
  display: flex;
  width: ${rem(200)};
  margin-top: ${rem(15)};
`;

export const FeaturedImage = styled.div`
  width: 100%;
  height: 100%;
  margin: 0;
  ${media.xs(css`
    transform: translateX(-25px);
  `)};
  ${media.lg(css`
    margin: ${rem(30)} 0 0 ${rem(100)};
    transform: none;
  `)};
`;

export const ImgHeader = styled.img`
  width: ${rem(120)};
  height: ${rem(120)};
  ${media.lg(css`
    width: ${rem(180)};
    height: ${rem(180)};
  `)};
`;
