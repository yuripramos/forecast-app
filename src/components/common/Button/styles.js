import React from "react";
import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { Link as RouterLink } from "react-router-dom";
import { lightBlue, white, green, green20 } from "../../../styles/settings";

export const ButtonTag = styled.button`
  background: ${lightBlue};
  font-family: "Helveticaneue Medium";
  border-radius: ${rem(4)};
  border: 0;
  cursor: pointer;
  display: inline-block;
  font-size: ${rem(12)};
  letter-spacing: ${rem(0.2)};
  line-height: ${rem(45)};
  outline: none;
  padding: 0 ${rem(20)};
  position: relative;
  color: ${white};
  text-align: center;
  text-decoration: none;
  transition: 0.2s ease-in-out color, 0.2s ease-in-out background,
    0.2s ease-in-out box-shadow, 0.2s ease-in-out border, 0.1s transform;
  white-space: nowrap;
  width: ${props => (props.width ? rem(props.width) : "100%")};
  user-select: none;
  opacity: ${props => (props.opacity && props.disabled ? props.opacity : 1)};

  ${media.md(css`
    width: ${props => (props.width ? rem(props.width) : "auto")};
  `)};

  ${media.lg(css`
    font-size: ${rem(13)};
    padding: 0 ${rem(28)};
    line-height: ${rem(48)};
  `)};

  :active {
    transform: translateY(${rem(1)});
  }

  ${({ isCallToAction }) =>
    isCallToAction &&
    css`
      background-color: ${green};
      box-shadow: 0 ${rem(2)} ${rem(7)} ${rem(2)} rgba(100, 100, 100, 0.3);
      color: ${white};
      :hover {
        background: ${green20};
      }
    `};
`;

export const RouterLinkTag = ButtonTag.withComponent(
  ({ isCallToAction, ...props }) => <RouterLink {...props} /> // eslint-disable-line
);
