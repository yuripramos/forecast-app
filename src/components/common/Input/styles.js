import styled, { css } from "styled-components";
import { rem } from "../../../styles/tools";
import { white, grey30, black50, darkBlue } from "../../../styles/settings";

export const InputField = styled.input`
  width: 30%;
  height: ${rem(25)};
  border-radius: ${rem(4)};
  background-color: ${white};
  border: solid ${rem(1)} ${grey30};
  letter-spacing: ${rem(0.8)};
  color: ${black50};
  padding: 0 ${rem(52)};
  transition: 0.3s padding, 0.3s border;

  ::-ms-clear {
    display: none !important;
  }

  :focus {
    outline: none;
    border: solid ${rem(1)} ${darkBlue};
  }
`;

export const InputWrapper = styled.div`
  width: 100%;
  display: inline-flex;
  position: relative;
  height: ${rem(26)};

  ${({ width }) =>
    width &&
    css`
      width: ${width}%;
    `};
`;

export const Label = styled.label`
  pointer-events: none;
  transition: 0.3s top, 0.2s color, 0.3s font-size;
  font-family: "Helveticaneue Medium";
  font-size: ${rem(16)};
  transform: translateY(5px);
`;
