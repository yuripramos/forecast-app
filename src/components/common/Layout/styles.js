import styled from "styled-components";
import { rem } from "../../../styles/tools";

export const Content = styled.div`
  min-height: calc(100vh - ${rem(500)});
`;
