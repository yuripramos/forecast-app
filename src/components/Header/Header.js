import React from "react";
import {
  HeaderWrapper,
  Title,
  SubTitle,
  ListIcons,
  IconWrapper,
  ImgHeader,
  FeaturedImage
} from "./styles";
import { Container, Row, Column } from "../../styles/grid";
import { white } from "../../styles/settings/index";
import { isResponsive } from "../../utils/getResolution";
import Icon from "../common/Icon";
import Hide from "../common/Hide";
import FeaturedHandImage from "../../assets/images/favicon.png";

function Header() {
  return (
    <HeaderWrapper>
      <Container>
        <Row>
          {!isResponsive() && (
            <Column width={1 / 2}>
              <FeaturedImage>
                <ImgHeader src={FeaturedHandImage} alt="trivia logo" />
              </FeaturedImage>
            </Column>
          )}
          <Column width={isResponsive() ? "auto" : "50%"}>
            <Title>Forecast</Title>
            <SubTitle> Future or past? </SubTitle>
            <Hide below="lg">
              <ListIcons>
                <IconWrapper>
                  <Icon
                    type="AndroidLogo"
                    height={33}
                    width={33}
                    color={white}
                  />
                </IconWrapper>
                <IconWrapper>
                  <Icon type="AppleLogo" height={33} width={33} />
                </IconWrapper>
                <IconWrapper>
                  <Icon type="MicrosoftLogo" height={33} width={33} />
                </IconWrapper>
              </ListIcons>
            </Hide>
          </Column>
        </Row>
      </Container>
    </HeaderWrapper>
  );
}

export default Header;
