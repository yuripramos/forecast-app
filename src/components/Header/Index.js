import { connect } from "redux-zero/react";

import Header from "./Header";
import Forecastactions from "../../actions/Forecast";

export default connect(
  ({ handleUserInput, search, clearSearch, isLoading }) => ({
    handleUserInput,
    search,
    clearSearch,
    isLoading
  }),
  Forecastactions
)(Header);
