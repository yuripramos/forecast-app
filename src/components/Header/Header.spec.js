import React from "react";
import { shallow, mount } from "enzyme";
import Header from "./Header";
import { FeaturedImage } from "./styles";
beforeEach(() => {
  jest.mock("../../utils/getResolution");
});

const isResponsive = require("../../utils/getResolution").isResponsive;

describe("Header component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<Header />)).toMatchSnapshot();
  });
  it("should render responsive header", () => {
    global.innerWidth = 500;
    expect(isResponsive()).toBe(true);
  });
  it("should render desktop header", () => {
    global.innerWidth = 1280;
    expect(!isResponsive()).toBe(false);
  });
  it("should return FeaturedImage Comp", () => {
    global.innerWidth = 1280;

    expect(mount(<Header />).find(FeaturedImage).length).toBe(0);
  });
});
