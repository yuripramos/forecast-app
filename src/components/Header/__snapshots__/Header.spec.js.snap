// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot with props 1`] = `
<styled.header>
  <Container>
    <Row>
      <Column
        width="auto"
      >
        <styled.h1>
          Trivia Game
        </styled.h1>
        <styled.h2>
           Welcome to the Trivia Challenge 
        </styled.h2>
        <Hide
          above={null}
          below="lg"
        >
          <styled.div>
            <styled.div>
              <Icon
                color="#FFFFFF"
                height={33}
                type="AndroidLogo"
                width={33}
              />
            </styled.div>
            <styled.div>
              <Icon
                color={null}
                height={33}
                type="AppleLogo"
                width={33}
              />
            </styled.div>
            <styled.div>
              <Icon
                color={null}
                height={33}
                type="MicrosoftLogo"
                width={33}
              />
            </styled.div>
          </styled.div>
        </Hide>
      </Column>
    </Row>
  </Container>
</styled.header>
`;
